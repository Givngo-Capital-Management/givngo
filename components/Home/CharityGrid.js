import React from 'react';
import { Divider } from 'semantic-ui-react';
import CharityPreview from '../Charity/CharityPreview';
import CharityGridHeader from './CharityGridHeader';
import './CharityGrid.scss';

export default function CharityGrid(props) {
  const { title, hideDivider } = props;
  return (
    <>
      <CharityGridHeader title={title} />
      <div className="charity-grid">
        <CharityPreview />
        <CharityPreview />
        <CharityPreview />
        <CharityPreview />
      </div>
      {hideDivider ? null : <Divider />}
    </>
  );
}
