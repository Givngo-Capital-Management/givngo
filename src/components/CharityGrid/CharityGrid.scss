.charity-grid {
  display: flex;
  flex-wrap: wrap;

  & > * {
    margin-left: 4px;
  }
}