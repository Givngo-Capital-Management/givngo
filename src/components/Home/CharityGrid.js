import React from 'react';
import { Divider } from 'semantic-ui-react';
import CharityPreview from '../Charity/CharityPreview';
import CharityGridHeader from './CharityGridHeader';
import styles from './CharityGrid.module.scss';

export default function CharityGrid(props) {
  const { title, hideDivider } = props;
  return (
    <>
      <CharityGridHeader title={title} />
      <div className={styles.grid}>
        <CharityPreview />
        <CharityPreview />
        <CharityPreview />
        <CharityPreview />
      </div>
      {hideDivider ? null : <Divider />}
    </>
  );
}
