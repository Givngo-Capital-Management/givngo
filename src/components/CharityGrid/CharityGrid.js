import React from 'react';
import './CharityGrid.scss';
import { Divider } from 'semantic-ui-react';
import CharityPreview from '../CharityPreview/CharityPreview';
import CharityGridHeader from './CharityGridHeader/CharityGridHeader';

export default function CharityGrid(props) {
  const { title, hideDivider } = props;
  return (
    <>
      <CharityGridHeader title={title} />
      <div className="charity-grid">
        <CharityPreview />
        <CharityPreview />
        <CharityPreview />
        <CharityPreview />
        <CharityPreview />
      </div>
      {hideDivider ? null : <Divider />}
    </>
  );
}
