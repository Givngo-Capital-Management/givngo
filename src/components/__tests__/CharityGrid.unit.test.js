import React from 'react';
import { shallow } from 'enzyme';
import CharityGrid from '../Home/CharityGrid';

describe('CharityGrid', () => {
  test('renders with title prop', () => {
    const wrapper = shallow(<CharityGrid title="Trending" charities={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders without divider', () => {
    const wrapper = shallow(<CharityGrid hideDivider charities={[]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
