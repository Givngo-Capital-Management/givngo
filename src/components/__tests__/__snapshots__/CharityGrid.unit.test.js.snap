// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharityGrid renders with title prop 1`] = `
<Fragment>
  <CharityGridHeader
    title="Trending"
  />
  <div
    className="grid"
  >
    <CharityPreview />
    <CharityPreview />
    <CharityPreview />
    <CharityPreview />
  </div>
  <Divider />
</Fragment>
`;

exports[`CharityGrid renders without divider 1`] = `
<Fragment>
  <CharityGridHeader />
  <div
    className="grid"
  >
    <CharityPreview />
    <CharityPreview />
    <CharityPreview />
    <CharityPreview />
  </div>
</Fragment>
`;

exports[`CharityGrid renders without props 1`] = `
<Fragment>
  <CharityGridHeader />
  <div
    className="grid"
  >
    <CharityPreview />
    <CharityPreview />
    <CharityPreview />
    <CharityPreview />
  </div>
  <Divider />
</Fragment>
`;
